tests/Application/Messenger/Handler/DonationUpsertedHandlerTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace MatchBot\Tests\Application\Messenger\Handler;
+
+use MatchBot\Application\Messenger\DonationUpserted;
+use MatchBot\Application\Messenger\Handler\DonationUpsertedHandler;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Tests\TestCase;
+use Prophecy\Argument;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+
+class DonationUpsertedHandlerTest extends TestCase
+{
+    use ProphecyTrait;
+
+    /** @var ObjectProphecy<DonationRepository>  */
+    private ObjectProphecy $donationRepositoryProphecy;
+
+    public function setUp(): void
+    {
+        $this->donationRepositoryProphecy = $this->prophesize(DonationRepository::class);
+    }
+
+    public function testItPushesOneDonationToSf(): void
+    {
+        $this->donationRepositoryProphecy->push(Argument::type(DonationUpserted::class), false)->shouldBeCalledOnce();
+
+        $sut = new DonationUpsertedHandler(
+            $this->donationRepositoryProphecy->reveal(),
+            new NullLogger(),
+        );
+
+        $sut->__invoke(self::someUpsertedMessage());
```

Comment:
Can we assert on the particular message we push, not just the type?

e.g.

```php
        $message = self::someUpsertedMessage();
        $this->donationRepositoryProphecy->push($message, false)->shouldBeCalledOnce();

        $sut = new DonationUpsertedHandler(
            $this->donationRepositoryProphecy->reveal(),
            new NullLogger(),
        );

        $sut->__invoke($message);
```

It doesn't really matter whether we make the same change or not in the next test.

Reply:
Done in e7d63af6c540d6c7b4bc2a9273a3bbbd50e74c66